src/AcceptanceTests/Receiving/When_receiving_a_message.cs

```diff
@@ -0,0 +1,67 @@
+﻿namespace NServiceBus.Transport.AzureServiceBus.AcceptanceTests.Sending.Receiving
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using NServiceBus.AcceptanceTests.EndpointTemplates;
+    using NUnit.Framework;
+    using Pipeline;
+
+    public class When_receiving_a_message
+    {
+        [Test]
+        public async Task Should_have_access_to_brokered_message_system_properties_via_the_context_bag()
+        {
+            await Scenario.Define<Context>()
+                .WithEndpoint<Endpoint>(b => b.When(
+                    (session, c) => session.SendLocal(new Message())))
+                .Done(c => c.LockedUntilUtcFromHandler == c.LockedUntilUtcFromBehavior)
+                .Run();
+        }
+
+        public class Context : ScenarioContext
+        {
+            public DateTime LockedUntilUtcFromHandler { get; set; }
+            public DateTime LockedUntilUtcFromBehavior { get; set; }
+        }
+
+        public class Endpoint : EndpointConfigurationBuilder
+        {
+            public Endpoint()
+            {
+                EndpointSetup<DefaultServer>(c => c.Pipeline.Register(b => new CheckContextForValidUntilUtc(b.Build<Context>()), "Behavior to validate context bag contains brokered message system properties"));
```

Comment:
There is an overload that allows to pass in the context directly then you don't need to do ` b.Build<Context>`  and can directly register the behavior with newing it up

Reply:
That's great, thank you.